dataService.js

```diff
@@ -0,0 +1,55 @@
+// Getting data from json files
+const categoryData = require('./src/data/categories.json');
+const productData = require('./src/data/products.json');
+
+// Maps
+let productMap = {};
+let categoryMap = {};
+let combinedProductMap = null;
+
+function getProducts() {
+    productData.products.map(function (val) {
+        productMap[val.id] = val;
+    });
+}
+
+function getCategories() {
+    categoryData.categories.map(function (val) {
+        categoryMap[val.id] = val.categoryName;
+    });
+}
+
+function combineProductsWithCategories() {
+
+    // Creating map of categories
+    getCategories();
+
+    combinedProductMap = productData.products.map(function (val) {
+        // Making deep copy to prevent changes in actual data
+        let productToBeReturned = JSON.parse(JSON.stringify(val));
+
+        // Adding categoryName
+        productToBeReturned.categoryName = categoryMap[productToBeReturned.categoryId];
+
+        return productToBeReturned;
+    });
+}
+
+function getCombinedProductMap() {
+
+    if (combinedProductMap == null) {
+        combineProductsWithCategories();
+    }
+
+    return combinedProductMap;
+}
+
+module.exports = {
+    getProducts: getProducts(),
+    getCategories: getCategories(),
+    combineProductsWithCategories : combineProductsWithCategories(),
+    getCombinedProductMap : getCombinedProductMap(),
+    productMap: productMap,
+    categoryMap: categoryMap,
+    combinedProductMap : combinedProductMap
```

Comment:
This shouldn't be exported - you already have `getCombinedProductMap`.

Reply:
Yeah, Done!